app/models/concerns/visible.rb

```diff
@@ -1,19 +1,11 @@
+# frozen_string_literal: true
+
 module Visible
   extend ActiveSupport::Concern
 
-  VALID_STATUSES = %w[public private archived]
-
-  included do
-    validates :status, inclusion: { in: VALID_STATUSES }
-  end
-
   class_methods do
-    def public_count
-      where(status: 'public').count
+    def published_article_count
+      Article.published.count
```

Comment:
Now this method is static in terms of working with models - it counts published only articles. Since you made it as concern you want to keep it model-agnostic, it should work with any model which has `published` method.
Try to change `Article` to `self`. Check if it works.